_posts/2016-02-24-publishing-data-with-odata-feeds.md

```diff
@@ -0,0 +1,100 @@
+---
+layout: post
+title: Publishing data with OData feeds
+categories:
+tags: [ReST, .NET, OData, MongoDB ]
+excerpt: How to publish readonly data sources with OData
+author: Sakis
+image:
+canonical:
+---
+## When reporting is a pain...
+
+Ever since I got into software business, a system's reporting capability has always been a pain either when it was the core of the system or being a nice to have feature. In our application, a distributed architecture comprised of several microservices supporting a workflow with different entry points, when the need for reporting became apparent heads started to scratch. And it wasn't the lack of skills that had us thinking about ways to expose the data to their stakeholders rather than the diversity and sometimes ambiguity of the reporting requirements (the "I want it ALL and I want it MY WAY" #agile data report).
+
+## Why OData?
+
+Why not OData? We use MongoDB for all data stores (it's a blasphemy to even mention SQL in our place) and our first thought was to use some sort of ODBC driver with SQL support to expose our _schemaless_ data. But then again, losing control of our system intrinsic processes by relying on black box components that came with a monthly bill, is enough incentive to stir our hacking/creative ingenuity.
+So, OData is free and open source (Microsoft says...). In a sentence, it's a web protocol that allows publishing data over HTTP along with a description (**metadata**) of their structure and their relations to other entities.
+
+Why bother with structure and references to other entities? **SHOW ME THE DATA!** Well, metadata information about the data are used by clients like Excel to reconstitute data schema in their own mysterious ways and hydrate/map the data on the other side of the wire. Excel is a buzzword in our business so it didn't take long to put two and two together.
```

Comment:
> metadata information about the data are used by clients

*Either*

metadata information is used by client

or

meta information about the data is used by the client